picasso/src/main/java/com/squareup/picasso3/SourceBufferingInputStream.java

```diff
@@ -33,33 +35,49 @@
   private final Buffer buffer;
   private long position;
   private long mark = -1;
+  private long markLimit = -1;
 
   SourceBufferingInputStream(BufferedSource source) {
     this.source = source;
     this.buffer = source.buffer();
   }
 
   private final Buffer temp = new Buffer();
+  // offset is the write offset in the dest array
   private int copyTo(byte[] sink, int offset, int byteCount) {
     // TODO replace this with https://github.com/square/okio/issues/362
-    buffer.copyTo(temp, offset, byteCount);
+    buffer.copyTo(temp, position, byteCount);
```

Comment:
I meant to circle back and fix this in Picasso, but I noticed this was a bug when trying to add `copyTo` to Okio. It becomes really annoying because we need two offsets in the API.